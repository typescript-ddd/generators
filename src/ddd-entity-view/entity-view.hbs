import { {{pascal entityName}} } from "../../domain/models";
{{#each properties}}
{{#contains "View" creator}}
import { {{../creator}} } from "./{{../creator}}";
{{else}}
{{#ifDefined creator}}
import { {{../creator}} } from "../../domain/models";
{{/ifDefined}}
{{/contains}}
{{/each}}
{{#each properties}}
{{#contains "View" creator}}
{{/contains}}
{{/each}}

export class {{pascal entityName}}View {
  private constructor({{lower entityName}}: {{pascal entityName}}) {
    this.id = {{lower entityName}}.getId().value;
    {{#each properties}}
    {{#ifDefined creator}}
    this.{{../name}} = {{../creator}}.create({{lower ../../entityName}}.get{{pascal ../name}}()){{#ifDefined ../value}}.{{../value}}{{/ifDefined}};
    {{else}}
    this.{{../name}} = {{lower ../../entityName}}.get{{pascal ../name}}(){{#ifDefined ../value}}.{{../value}}{{/ifDefined}};
    {{/ifDefined}}
    {{/each}}
  }

  public readonly id: string;

  {{#each properties}}
  public readonly {{name}}: {{type}};
  {{/each}}
}