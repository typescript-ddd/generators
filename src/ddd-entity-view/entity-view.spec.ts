import { generateEntityView } from "./entity-view.generator";
import { EntityViewTemplateData } from "./entity-view.types";

describe("EntityView", () => {
  it("should render a created entity view with view property", () => {
    // Arrange
    const data: EntityViewTemplateData = {
      entityName: "User",
      properties: [
        { name: "name", type: "string", value: "value" },
        { name: "email", type: "EmailView", creator: "EmailView" },
      ],
    };

    // Act
    const result = generateEntityView(data);

    // Assert
    expect(result).toMatchSnapshot();
  });

  it("should render a created entity view with view property value", () => {
    // Arrange
    const data: EntityViewTemplateData = {
      entityName: "User",
      properties: [
        { name: "name", type: "string", value: "value" },
        {
          name: "email",
          type: "string",
          creator: "EmailView",
          value: "emailAddress",
        },
      ],
    };

    // Act
    const result = generateEntityView(data);

    // Assert
    expect(result).toMatchSnapshot();
  });
});
