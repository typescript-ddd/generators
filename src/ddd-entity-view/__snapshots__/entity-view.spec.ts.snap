// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntityView should render a created entity view with view property 1`] = `
"import { User } from "../../domain/models";
import { EmailView } from "./EmailView";

export class UserView {
  private constructor(user: User) {
    this.id = user.getId().value;
    this.name = user.getName().value;
    this.email = EmailView.create(user.getEmail());
  }

  public readonly id: string;

  public readonly name: string;
  public readonly email: EmailView;
}"
`;

exports[`EntityView should render a created entity view with view property value 1`] = `
"import { User } from "../../domain/models";
import { EmailView } from "./EmailView";

export class UserView {
  private constructor(user: User) {
    this.id = user.getId().value;
    this.name = user.getName().value;
    this.email = EmailView.create(user.getEmail()).emailAddress;
  }

  public readonly id: string;

  public readonly name: string;
  public readonly email: string;
}"
`;
